logistics_planning_invoicing/models/logistics_schedule.py

```diff
@@ -0,0 +1,73 @@
+# © 2024 Solvos Consultoría Informática (<http://www.solvos.es>)
+# License LGPL-3.0 (https://www.gnu.org/licenses/lgpl-3.0.html)
+
+from odoo import models, fields, _
+from odoo.exceptions import ValidationError
+
+
+class LogisticsSchedule(models.Model):
+    _inherit = 'logistics.schedule'
+
+    account_move_line_id = fields.Many2one('account.move.line')
+    account_move_id = fields.Many2one('account.move', related='account_move_line_id.move_id', string='Invoice')
+
+    def is_invoiceable(self):
+        return any(not record.account_move_line_id or record.state == 'done' for record in self)
+
+    def action_create_invoice_wizard(self):
+        Wizard = self.env['logistics.schedule.account.move.wizard']
+        partner_id = self.env['logistics.schedule'].browse(self.env.context.get("active_ids")).mapped('partner_id')
+        if len(partner_id) > 1:
+            raise ValidationError(_("You cannot select multiple records with different partners."))
+        new = Wizard.create({
+            "logistics_schedule_ids": self.env.context.get("active_ids"),
+            'carrier_id': partner_id.id
+        })
+        return {
+            'name': _('Create Invoice'),
+            'res_model': 'logistics.schedule.account.move.wizard',
+            'view_mode': 'form',
+            'view_type': 'form',
+            'res_id': new.id,
+            'target': 'new',
+            'type': 'ir.actions.act_window',
+        }
+
+    def action_logistics_schedule_create_invoice(self):
```

Comment:
I don't understand this method, with a `return` within the `for` statement. Where is it used?